File: Demo3/app/src/main/java/com/hiscene/demo3/MainActivity.java
package com.hiscene.demo3;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

import com.hiscene.demo3.eventbus.EventBus;
import com.hiscene.demo3.eventbus.Subscribe;
import com.hiscene.demo3.eventbus.ThreadMode;
import com.hiscene.demo3.eventbus.UserInfo;

public class MainActivity extends AppCompatActivity {
    private static final String TAG = "MainActivityTAG";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        //告诉EventBus将本类的方法都放到eventbus中管理
        try {
            EventBus.getDefault().register(this);
        } catch (Exception e) {
            e.printStackTrace();
        }

        findViewById(R.id.btn_goto).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(MainActivity.this,Main2Activity.class);
                startActivity(intent);
            }
        });

    }

    @Subscribe(threadMode = ThreadMode.MAIN)
    public void aa(UserInfo userInfo){
        Log.d(TAG, "aa: "+userInfo.toString());
    }

    @Subscribe(threadMode = ThreadMode.BACKGROUND)
    public void bb(String name){

    }
}

File: Demo2/architecturecomponent/src/main/java/com/hiscene/architecturecomponent/MainActivity.kt
package com.hiscene.architecturecomponent

import android.os.Bundle
import android.text.TextUtils
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.ViewModelProviders
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    lateinit var model: NameViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initView()
    }

    private fun initView() {
        model = ViewModelProvider(this).get(NameViewModel::class.java)

        model.getUserInfo().observe(this, Observer {
            if (it.isNotEmpty())
            it[0].run {
                txt_info.text = "name=$name,id=$id,age=$age"
            }
        })

        btn_sure.setOnClickListener {
            if (TextUtils.isEmpty(edit_age.text) || TextUtils.isEmpty(edit_name.text) || TextUtils.isEmpty(edit_no.text)) {
                toast("请输入所有信息")
            } else {
                var userInfo = UserInfo(edit_no.text.toString().toInt(), edit_name.text.toString(), edit_age.text.toString().toInt())
                model.insert(userInfo)
            }
        }
    }

    private fun toast(msg: String) {
        Toast.makeText(this, msg, Toast.LENGTH_SHORT).show()
    }
}
